pyanaconda/core/path.py

```diff
@@ -0,0 +1,154 @@
+
+    The path defined by conf.target.system_root will be bind mounted at the given
+    path, so conf.target.system_root can be used to access the root of the new OS.
+
+    We always call it after the root device is mounted at conf.target.physical_root
+    to set the physical root as the current system root.
+
+    Then, it can be used by Payload subclasses which install operating systems to
+    non-default roots.
+
+    If the given path is None, then conf.target.system_root is only unmounted.
+
+    :param path: the new OS root path or None
+    :type path: str or None
+    """
+    from pyanaconda.core.util import execWithRedirect
+
+    sysroot = conf.target.system_root
+
+    if sysroot == path:
+        return
+
+    # Unmount the mount point if necessary.
+    rc = execWithRedirect("findmnt", ["-rn", sysroot])
+
+    if rc == 0:
+        execWithRedirect("mount", ["--make-rprivate", sysroot])
+        execWithRedirect("umount", ["--recursive", sysroot])
+
+    if not path:
+        return
+
+    # Create a directory for the mount point.
+    if not os.path.exists(sysroot):
+        mkdirChain(sysroot)
+
+    # Mount the mount point.
+    rc = execWithRedirect("mount", ["--rbind", path, sysroot])
+
+    if rc != 0:
+        raise OSError("Failed to mount sysroot to {}.".format(path))
+
+
+def mkdirChain(directory):
+    """ Make a directory and all of its parents. Don't fail if part or
+        of it already exists.
+
+        :param str directory: The directory path to create
+    """
+
+    os.makedirs(directory, 0o755, exist_ok=True)
+
+
+def dir_tree_map(root, func, files=True, dirs=True):
+    """Apply a function to all paths under a directory.
+
+    Apply the given function to all files and directories in the directory tree
+    under the given root directory.
+
+    :param root: root of the directory tree the function should be mapped to
+    :type root: str
+    :param func: a function taking the directory/file path
+    :type func: path -> None
+    :param files: whether to apply the function to the files in the dir. tree
+    :type files: bool
+    :param dirs: whether to apply the function to the directories in the dir. tree
+    :type dirs: bool
+
+    TODO: allow using globs and thus more trees?
+    """
+
+    for (dir_ent, _dir_items, file_items) in os.walk(root):
+        if dirs:
+            # try to call the function on the directory entry
+            try:
+                func(dir_ent)
+            except OSError:
+                pass
+
+        if files:
+            # try to call the function on the files in the directory entry
+            for file_ent in (os.path.join(dir_ent, f) for f in file_items):
+                try:
+                    func(file_ent)
+                except OSError:
+                    pass
+
+        # directories under the directory entry will appear as directory entries in the loop
+
+
+def chown_dir_tree(root, uid, gid, from_uid_only=None, from_gid_only=None):
```

Comment:
I would move the `chown_dir_tree` and `dir_tree_map` functions to `pyanaconda.core.users` and make them private. Please, add `FIXME: Use 'chown -hR' instead of this function.` to their docstings.

Reply:
I am not sure about the FIXME, `chown` will not use the logic that is in the function. However the rest seems a good idea, although there are more uses. Regardless, this can be trimmed significantly, half code paths are unused...

Reply:
About the FIXME, I was googling how to change the owner of a home directory and this is what I have found. I didn't have time to investigate why are our functions are so complicated and the commit messages are not very helpful. Feel free to reword the text, but please keep there the command as a hint.